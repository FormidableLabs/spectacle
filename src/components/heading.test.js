import React from 'react';
import { mount } from 'enzyme';
import Heading from './heading';

describe('<Heading />', () => {
  test('should render correctly.', () => {
    const context = {
      styles: {
        components: {
          heading: {
            h2: {
              color: '#ff0'
            }
          }
        }
      }
    };
    const wrapper = mount(<Heading size={2}>Hi There!</Heading>, { context });
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the fit configuration correctly.', () => {
    const context = { styles: { components: { heading: {} } } };
    const wrapper = mount(<Heading fit>This Header Fits!</Heading>, {
      context
    });
    expect(wrapper).toMatchSnapshot();
  });
});
