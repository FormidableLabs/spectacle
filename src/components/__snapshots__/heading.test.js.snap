// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render correctly. 1`] = `
<Heading
  lineHeight={1}
  size={2}
>
  <H2
    data-radium={true}
    style={
      Object {
        "color": "#ff0",
        "lineHeight": 1,
      }
    }
  >
    Hi There!
  </H2>
</Heading>
`;

exports[`<Heading /> should render the fit configuration correctly. 1`] = `
<Heading
  fit={true}
  lineHeight={1}
  size={1}
>
  <div
    data-radium={true}
    style={
      Object {
        "display": "block",
        "height": 0,
        "width": "100%",
      }
    }
  >
    <span
      data-radium={true}
      style={
        Object {
          "display": "block",
          "fontSize": 16,
          "lineHeight": 1,
          "margin": "0",
          "padding": "0",
          "transform": "scale(NaN)",
          "transformOrigin": "center top",
        }
      }
    >
      This Header Fits!
    </span>
  </div>
</Heading>
`;
